addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -307,6 +336,47 @@ export class LinkPlugin extends Plugin {
                 return;
             }
             this.overlay.close();
+        } else if (!selection.isCollapsed) {
+            const selectedNodes = this.shared.getSelectedNodes();
+            const imageNode = selectedNodes.find((node) => node.tagName === "IMG");
+            if (imageNode && imageNode.parentNode.tagName === "A") {
+                if (this.linkElement !== imageNode.parentElement) {
+                    this.overlay.close();
+                    this.removeCurrentLinkIfEmtpy();
+                }
+                this.linkElement = imageNode.parentElement;
+                const props = {
+                    linkEl: this.linkElement,
+                    onApply: (url, _) => {
+                        this.linkElement.href = url;
+                        this.shared.setCursorEnd(this.linkElement);
+                        this.dispatch("ADD_STEP");
+                        this.removeCurrentLinkIfEmtpy();
+                    },
+                    onRemove: () => {
+                        this.removeLink();
+                        this.overlay.close();
+                        this.dispatch("ADD_STEP");
+                    },
+                    onCopy: () => {
+                        this.overlay.close();
+                    },
+                    onClose: () => {
+                        this.overlay.close();
+                    },
+                    getInternalMetaData: this.getInternalMetaData,
+                    getExternalMetaData: this.getExternalMetaData,
+                    isImage: true,
+                };
+                // close the overlay to change the target
+                if (this.overlay.isOpen) {
+                    this.overlay.close();
+                }
+                // pass the link element to overlay to prevent position change
+                this.overlay.open({ target: imageNode, props });
```

Comment:
why do you need to close and open after ?

Reply:
I closed the opened overlay cause sometimes the overlay service updates the position based on the current selection and the target is changed to the link element (`<a> image</a>`). Then the popover is not at the bottom of the image